powermon/formats/abstractformat.py

```diff
@@ -0,0 +1,85 @@
+from abc import ABC, abstractmethod
+import logging
+import re
+from enum import StrEnum, auto
+
+# from time import sleep
+log = logging.getLogger("Formatter")
+
+class FormatterType(StrEnum):
+    HASS = auto()
+    HTMLTABLE = auto()
+    RAW = auto()
+    SIMPLE = auto()
+    TABLE = auto()
+    TOPICS = auto()
+
+class AbstractFormat(ABC):
+
+    def __init__(self, formatConfig):
+        self.remove_spaces = formatConfig.get("remove_spaces", True)
+        self.keep_case = formatConfig.get("keep_case", False)
+        
+        self._keyFilter = None
+        _keyFilterString = formatConfig.get("filter", None)
+        if _keyFilterString is not None:
+            self._keyFilter = re.compile(_keyFilterString)
+
+        self._keyExclusionfilter = None
+        _keyExclusionFilterString = formatConfig.get("excl_filter", None)
+        if _keyExclusionFilterString is not None:
+            self._keyExclusionfilter = re.compile(_keyExclusionFilterString)
+
+   
+    @abstractmethod
+    def format(self, data):
+        pass
+
+    #Override this if the format sends multiple messages
+    def sendsMultipleMessages(self) -> bool:
+        return False
+
+    def formatAndFilterData(self, data):
+        #TODO: should we make data a proper object so it's easy to get the data we want?
```

Comment:
Was just thinking this and yes I think we need a 'response' object (or similar) and embed the complexity there